### ISSUE TITLE
Lua filter — type_int_key for large numbers doesn't work

### ISSUE THREAD

**Issue by jamesgoodhouse** [fluent/fluent-bit#1099]
## Bug Report

**Describe the bug**

If a value is too large for a field that has type_int_key configured for it, the number that is returned is incorrect.

**To Reproduce**

Setup a filter that uses a Lua script:

```
[FILTER]
    Name lua
    Match *
    Script test.lua
    call test
    type_int_key big_number
```

Create a Lua function that involves a very big number.

```
function test(tag, timestamp, record)
  local new_record = {}
  
  new_record.big_number = 1549661409000

  return 1, timestamp, new_record
end
```

Observe in your logs that the number is not what is expected.

```
{"big_number": -2147483648}
```

**Expected behavior**

I expect the following log:

```
{"big_number": 1549661409000}
```

**Screenshots**

N/A

**Your Environment**

* Version used: `1.0.4`
* Configuration: Running in Docker using `fluent/fluent-bit:1.0.4`
* Operating System and version: MacOS 10.13.6

**Additional context**

We have an endpoint that is expecting a timestamp as an integer that represents the epoch time in milliseconds.

**Comment by jesk78** [comment]
Isn't the purpose of _type_int_key_ to use integer instead of double? So not using it doesn't help?

**Comment by bluebike** [comment]
If I understand right. 
Luajit-2.X used by fluent-bit is based Lua 5.1, which has only 32bit integers. 
(Lua 5.3 uses 64bit integers by default)

**Comment by jesk78** [comment]
No idea, but maybe it's ok to just cast to int64_t instead of int in lua.c:
```
<< msgpack_pack_int64(pck, (int)lua_tonumber(l, -1));

>> msgpack_pack_int64(pck, (int64_t)lua_tonumber(l, -1));
```

**Comment by jamesgoodhouse** [comment]
@jesk78 Does that work with the current version of Lua that's bundled with fluent-bit?

**Comment by edsiper** [comment]
@jesk78 your fix is right, that cast in the current code truncates the value. Can you submit a PR with the fix ?

**Comment by jesk78** [comment]
Here we go...

#1103

**Comment by edsiper** [comment]
thanks, merged.

note: Commits must be prefixed, changed to "filter_lua: fix casting of int64 value (#1103)".

issue fixed.
